[#69101] create UI for artifact export in project initiation request

Kharonus opened the pull request:
# Ticket
[#69101](https://community.openproject.org/work_packages/69101)

# What are you trying to accomplish?
- show UI for artifact export configuration
- no project settings update - yet
- no form initialization with project settings - yet

# What approach did you choose and why?
- Add form and view components
- add logic to list correct file storages
- add settings for the configurable attributes

# Hint

- As Mir is currently working on the creation wizard controller, I didn't want to implement the update in parallel. Instead I forwarded some feedback and expectations.

mereghost commented on app/forms/projects/settings/creation_wizard/export_form.rb:
```diff
@@ -0,0 +1,97 @@
+        end
+
+        def file_storages
+          @file_storages ||= model.project_storages
```
🟡 WDYT about adding a named scope for this, like `automatically_managed_project_storages`, for clarifying the intent of the query.

Kharonus replied:
sounds good, will do

Kharonus replied:
actually a scope already existed, I simply missed to use it.

Kharonus committed:
[#69101] add dynamic form behaviour to export artifact

- use existing scope
- fix feature test click targets

NobodysNightmare commented on app/forms/projects/settings/creation_wizard/export_form.rb:
```diff
@@ -0,0 +1,97 @@
+          end
+
+          if file_storages.any?
+            f.select_list(
```
:yellow_circle: Not sure how elaborate our designs for this are, but other comparable UIs (input below radio buttons, where the input is only needed for one radio selection) would hide the select-list when the radio selection is on "attachments".

We repeated this pattern so many times, that I am wondering, whether we should extract it into a dedicated stimulus controller by now. Prior art:

* subject configuration dialog
* audience selection in Nextcloud storage config
* create SCIM client (here a select decides visibility, not a radio input)

Kharonus replied:
Parimal says, we should have same dynamic behaviour -> added

NobodysNightmare commented on app/forms/projects/settings/creation_wizard/export_form.rb:
```diff
@@ -0,0 +1,97 @@
+          if file_storages.any?
+            label
+          else
+            "#{label} (#{I18n.t('projects.settings.creation_wizard.export.unavailable')})"
```
:yellow_circle: Not sure what the designs say about this, but I don't know of other radio buttons that are disabled and then receive additional text that points out they are disabled.

IMO this text is useless: It tells us what we already know (the radio is disabled), without telling us anything interesting (why it is disabled).

My preference would be to always show the same label here.

Kharonus replied:
mine, too, I just followed the figma ^^

Kharonus replied:
Parimal says, that the suffix is needed, due to being a very simple try of explaining WHY it is disabled.

NobodysNightmare commented on app/models/projects/creation_wizard.rb:
```diff
@@ -33,5 +33,7 @@ module Projects::CreationWizard
 
   included do
     store_attribute :settings, :project_creation_wizard_enabled, :boolean
+    store_attribute :settings, :project_creation_wizard_pdf_export, :string, default: "attachment"
```
:green_circle: Should we also suffix this?

* `project_creation_wizard_pdf_export_type`
* `project_creation_wizard_pdf_export_destination`

Kharonus replied:
yes, can do

NobodysNightmare commented on app/components/projects/settings/creation_wizard/export_component.html.erb:
```diff
@@ -0,0 +1,37 @@
+  settings_primer_form_with(
+    model: @project,
+    id: "project_creation_wizard_export_artifact_form",
+    url: "https://todo.update-route.controller",
```
:green_circle: I'd avoid adding such absolute URLs as placeholders ever. Imagine that we forget about this due to shifting priorities or whatever. Suddenly there's code inside OpenProject that points to a random URL that someone could buy (once the `.controller` TLD is available).

The safer alternative (that remains a placeholder) would be to make sure we remain on the same server, e.g.

```suggestion
    url: "/todo/update/route",
```

Kharonus replied:
true enough 😂

NobodysNightmare reviewed (approved):
While I have :yellow_circle: remarks, I am pretty sure that the answer will be in the designs that you got.

Still leaving them for your consideration.

Kharonus committed:
[#69101] create UI for artifact export in project initiation request

- https://community.openproject.org/work_packages/69101
- Add form and view components
- add logic to list correct file storages
- add settings for the configurable attributes
